keanu-project/src/main/java/io/improbable/keanu/util/io/ProtobufLoader.java

```diff
@@ -59,9 +61,14 @@ public BayesianNetwork loadNetwork(KeanuSavedBayesNet.Model parsedModel) {
 
     @Override
     public void loadValue(DoubleVertex vertex) {
-        KeanuSavedBayesNet.VertexValue value = savedValues.get(vertex);
+        KeanuSavedBayesNet.StoredValue valueInformation = savedValues.get(vertex);
+        KeanuSavedBayesNet.VertexValue value = valueInformation.getValue();
         DoubleTensor tensor = extractDoubleValue(value);
-        vertex.setValue(tensor);
+        if (valueInformation.getIsObserved()) {
```

Comment:
Could this if block go into it's own method as it is repeated in each `loadValue`?  Something like `setValueAndObservedState`

Reply:
Hmm, ok, didn't really think it was worth doing, as it makes it less clear imo and encapsulates very little code. But it'd definitely does make it easier to extend in the future, so will rewrite.